test/framework/kogitoapp.go

```diff
@@ -211,6 +214,42 @@ func setupBuildImageStreams(kogitoApp *v1alpha1.KogitoApp) {
 	// If none, let the operator manage
 	kogitoApp.Spec.Build.ImageS2ITag = config.GetBuildS2IImageStreamTag()
 	kogitoApp.Spec.Build.ImageRuntimeTag = config.GetBuildRuntimeImageStreamTag()
+	if len(config.GetBuildImageRegistry()) > 0 || len(config.GetBuildImageNamespace()) > 0 {
+		overwriteBuildImageRuntimeIfNotSet(kogitoApp)
+		overwriteBuildImageS2IIfNotSet(kogitoApp)
+	}
+
 	// If "KOGITO_BUILD_IMAGE_VERSION" is defined, it's taken into account, otherwise set the current version
 	kogitoApp.Spec.Build.ImageVersion = config.GetBuildImageVersion()
 }
+
+func overwriteBuildImageRuntimeIfNotSet(kogitoApp *v1alpha1.KogitoApp) {
```

Comment:
What about adjusting this function to something like
```suggestion
func getBuildRuntimeImageStreamTag(kogitoApp *v1alpha1.KogitoApp) string {
```
And here you would completely resolve the tag?
The logic will stay the same, just in case` len(config.GetBuildRuntimeImageStreamTag()) > 0` you can return `config.GetBuildRuntimeImageStreamTag()`.
This would IMHO be more readable.

Reply:
I refactor this change to try making more readable. Let me know your thoughts now.
Thanks!